internal/engine/planner.go

```diff
@@ -30,18 +29,14 @@ func PlannerEvaluateRolePolicy(ctx context.Context, tctx tracer.Context, evaluat
 	}
 	result, err := evaluator.Evaluate(ctx, tctx, &checkInput)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	if len(result.ValidationErrors) > 0 {
-		return nil, nil, errors.New("role policies produced validation errors") // this shouldn't happen as role policies doesn't evaluate result.
+		return nil, errors.New("role policies produced validation errors") // this shouldn't happen as role policies doesn't evaluate result.
 	}
-	if rpe, ok := evaluator.(*rolePolicyEvaluator); ok {
-		roles := make([]string, 0, len(rpe.policies))
-		for _, policy := range rpe.policies {
-			roles = append(roles, policy.Role)
-		}
-		return result, internal.ToSet(roles), nil
+	if _, ok := evaluator.(*rolePolicyEvaluator); ok {
```

Comment:
Can this be anything else? If so, shouldn't we do this check at the beginning of the function? 

Why not just change the `evaluator` type to `*rolePolicyEvaluator`?